{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly_ID</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Chromosome_ID</th>\n",
       "      <th>Plasmid_ID</th>\n",
       "      <th>Chr_len</th>\n",
       "      <th>Plasmid_len</th>\n",
       "      <th>Chr_GC</th>\n",
       "      <th>Plasmid_GC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_023703895.1_ASM2370389v1_genomic.fna.gz</td>\n",
       "      <td>KLEBSIELLA PNEUMONIAE</td>\n",
       "      <td>CP077812.1</td>\n",
       "      <td>CP077813.1</td>\n",
       "      <td>5167508</td>\n",
       "      <td>54288</td>\n",
       "      <td>0.575169</td>\n",
       "      <td>0.527888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100133.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>189346</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.638836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100134.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>128913</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.614430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100135.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>104740</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.636500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_012971805.2_ASM1297180v2_genomic.fna.gz</td>\n",
       "      <td>RHODOCOCCUS OPACUS</td>\n",
       "      <td>CP128998.1,CP128999.1</td>\n",
       "      <td>CP129000.1</td>\n",
       "      <td>9807637</td>\n",
       "      <td>54121</td>\n",
       "      <td>0.669073</td>\n",
       "      <td>0.667671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50871</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014366.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>3125</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.287360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50872</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014367.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>27068</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.305231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50873</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098817.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>44138</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.300580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50874</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098818.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>39504</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50875</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098819.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>22579</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289473</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50876 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Assembly_ID                Taxonomy  \\\n",
       "0      GCA_023703895.1_ASM2370389v1_genomic.fna.gz   KLEBSIELLA PNEUMONIAE   \n",
       "1      GCF_024662395.1_ASM2466239v1_genomic.fna.gz     BURKHOLDERIA GLUMAE   \n",
       "2      GCF_024662395.1_ASM2466239v1_genomic.fna.gz     BURKHOLDERIA GLUMAE   \n",
       "3      GCF_024662395.1_ASM2466239v1_genomic.fna.gz     BURKHOLDERIA GLUMAE   \n",
       "4      GCA_012971805.2_ASM1297180v2_genomic.fna.gz      RHODOCOCCUS OPACUS   \n",
       "...                                            ...                     ...   \n",
       "50871   GCF_003595385.1_ASM359538v1_genomic.fna.gz   STAPHYLOCOCCUS AUREUS   \n",
       "50872   GCF_003595385.1_ASM359538v1_genomic.fna.gz   STAPHYLOCOCCUS AUREUS   \n",
       "50873  GCF_039640675.1_ASM3964067v1_genomic.fna.gz  STAPHYLOCOCCUS CAPITIS   \n",
       "50874  GCF_039640675.1_ASM3964067v1_genomic.fna.gz  STAPHYLOCOCCUS CAPITIS   \n",
       "50875  GCF_039640675.1_ASM3964067v1_genomic.fna.gz  STAPHYLOCOCCUS CAPITIS   \n",
       "\n",
       "                     Chromosome_ID     Plasmid_ID  Chr_len  Plasmid_len  \\\n",
       "0                       CP077812.1     CP077813.1  5167508        54288   \n",
       "1      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100133.1  6558951       189346   \n",
       "2      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100134.1  6558951       128913   \n",
       "3      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100135.1  6558951       104740   \n",
       "4            CP128998.1,CP128999.1     CP129000.1  9807637        54121   \n",
       "...                            ...            ...      ...          ...   \n",
       "50871                NZ_CP014365.1  NZ_CP014366.1  2904958         3125   \n",
       "50872                NZ_CP014365.1  NZ_CP014367.1  2904958        27068   \n",
       "50873                NZ_CP098816.1  NZ_CP098817.1  2443214        44138   \n",
       "50874                NZ_CP098816.1  NZ_CP098818.1  2443214        39504   \n",
       "50875                NZ_CP098816.1  NZ_CP098819.1  2443214        22579   \n",
       "\n",
       "         Chr_GC  Plasmid_GC  \n",
       "0      0.575169    0.527888  \n",
       "1      0.683257    0.638836  \n",
       "2      0.683257    0.614430  \n",
       "3      0.683257    0.636500  \n",
       "4      0.669073    0.667671  \n",
       "...         ...         ...  \n",
       "50871  0.327958    0.287360  \n",
       "50872  0.327958    0.305231  \n",
       "50873  0.330543    0.300580  \n",
       "50874  0.330543    0.289464  \n",
       "50875  0.330543    0.289473  \n",
       "\n",
       "[50876 rows x 8 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load pairwise dataset\n",
    "fn = '/fs/nexus-scratch/rhaworth/plasmid/pairs.txt'\n",
    "data = pd.read_csv(fn, sep='\\t')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Seq_type</th>\n",
       "      <th>SeqID</th>\n",
       "      <th>AAAA</th>\n",
       "      <th>AAAC</th>\n",
       "      <th>AAAG</th>\n",
       "      <th>AAAT</th>\n",
       "      <th>AACA</th>\n",
       "      <th>AACC</th>\n",
       "      <th>AACG</th>\n",
       "      <th>...</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TACA</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TATA</th>\n",
       "      <th>TCAA</th>\n",
       "      <th>TCCA</th>\n",
       "      <th>TCGA</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGCA</th>\n",
       "      <th>TTAA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic</td>\n",
       "      <td>90881.0</td>\n",
       "      <td>42325.0</td>\n",
       "      <td>60895.0</td>\n",
       "      <td>68917.0</td>\n",
       "      <td>43978.0</td>\n",
       "      <td>24963.0</td>\n",
       "      <td>27889.0</td>\n",
       "      <td>...</td>\n",
       "      <td>67461.0</td>\n",
       "      <td>33931.0</td>\n",
       "      <td>31822.0</td>\n",
       "      <td>17586.0</td>\n",
       "      <td>52989.0</td>\n",
       "      <td>30674.0</td>\n",
       "      <td>9695.0</td>\n",
       "      <td>59606.0</td>\n",
       "      <td>17647.0</td>\n",
       "      <td>31864.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000005845.2_ASM584v2_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GCA_000005845.2_ASM584v2_genomic</td>\n",
       "      <td>70767.0</td>\n",
       "      <td>50477.0</td>\n",
       "      <td>45989.0</td>\n",
       "      <td>51593.0</td>\n",
       "      <td>43674.0</td>\n",
       "      <td>40859.0</td>\n",
       "      <td>48632.0</td>\n",
       "      <td>...</td>\n",
       "      <td>44665.0</td>\n",
       "      <td>21282.0</td>\n",
       "      <td>11307.0</td>\n",
       "      <td>9081.0</td>\n",
       "      <td>38771.0</td>\n",
       "      <td>33770.0</td>\n",
       "      <td>15462.0</td>\n",
       "      <td>51762.0</td>\n",
       "      <td>19768.0</td>\n",
       "      <td>21209.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000006605.1_ASM660v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GCA_000006605.1_ASM660v1_genomic</td>\n",
       "      <td>10309.0</td>\n",
       "      <td>11534.0</td>\n",
       "      <td>12081.0</td>\n",
       "      <td>6826.0</td>\n",
       "      <td>13481.0</td>\n",
       "      <td>18306.0</td>\n",
       "      <td>17508.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5623.0</td>\n",
       "      <td>7450.0</td>\n",
       "      <td>7014.0</td>\n",
       "      <td>1039.0</td>\n",
       "      <td>12658.0</td>\n",
       "      <td>24371.0</td>\n",
       "      <td>12682.0</td>\n",
       "      <td>15320.0</td>\n",
       "      <td>9582.0</td>\n",
       "      <td>2712.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000006625.1_ASM662v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GCA_000006625.1_ASM662v1_genomic</td>\n",
       "      <td>43692.0</td>\n",
       "      <td>13073.0</td>\n",
       "      <td>11309.0</td>\n",
       "      <td>33335.0</td>\n",
       "      <td>13659.0</td>\n",
       "      <td>4502.0</td>\n",
       "      <td>3782.0</td>\n",
       "      <td>...</td>\n",
       "      <td>33630.0</td>\n",
       "      <td>5964.0</td>\n",
       "      <td>5654.0</td>\n",
       "      <td>5887.0</td>\n",
       "      <td>13983.0</td>\n",
       "      <td>3824.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>12956.0</td>\n",
       "      <td>2571.0</td>\n",
       "      <td>16985.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000006645.1_ASM664v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GCA_000006645.1_ASM664v1_genomic</td>\n",
       "      <td>75622.0</td>\n",
       "      <td>47896.0</td>\n",
       "      <td>43794.0</td>\n",
       "      <td>60630.0</td>\n",
       "      <td>46161.0</td>\n",
       "      <td>43690.0</td>\n",
       "      <td>39865.0</td>\n",
       "      <td>...</td>\n",
       "      <td>55190.0</td>\n",
       "      <td>23874.0</td>\n",
       "      <td>17241.0</td>\n",
       "      <td>13213.0</td>\n",
       "      <td>48290.0</td>\n",
       "      <td>32235.0</td>\n",
       "      <td>11718.0</td>\n",
       "      <td>51982.0</td>\n",
       "      <td>18633.0</td>\n",
       "      <td>23183.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91820</th>\n",
       "      <td>GCF_964199155.1_Zagreb009_hybrid_assembly.fa.g...</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>NZ_OZ111549.1</td>\n",
       "      <td>79.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91821</th>\n",
       "      <td>GCF_964205035.1_18MA0850_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>NZ_OZ123584.1</td>\n",
       "      <td>131.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>370.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>...</td>\n",
       "      <td>36.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>267.0</td>\n",
       "      <td>294.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91822</th>\n",
       "      <td>GCF_964205035.1_18MA0850_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>NZ_OZ123585.1</td>\n",
       "      <td>79.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91823</th>\n",
       "      <td>GCF_964205045.1_18MA0854_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>NZ_OZ123587.1</td>\n",
       "      <td>119.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>224.0</td>\n",
       "      <td>315.0</td>\n",
       "      <td>356.0</td>\n",
       "      <td>...</td>\n",
       "      <td>28.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>233.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91824</th>\n",
       "      <td>GCF_964205045.1_18MA0854_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>NZ_OZ123588.1</td>\n",
       "      <td>79.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91825 rows Ã— 139 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Assembly    Seq_type  \\\n",
       "0                       GCA_000005825.2_ASM582v2_genomic  chromosome   \n",
       "1                       GCA_000005845.2_ASM584v2_genomic  chromosome   \n",
       "2                       GCA_000006605.1_ASM660v1_genomic  chromosome   \n",
       "3                       GCA_000006625.1_ASM662v1_genomic  chromosome   \n",
       "4                       GCA_000006645.1_ASM664v1_genomic  chromosome   \n",
       "...                                                  ...         ...   \n",
       "91820  GCF_964199155.1_Zagreb009_hybrid_assembly.fa.g...     plasmid   \n",
       "91821                   GCF_964205035.1_18MA0850_genomic     plasmid   \n",
       "91822                   GCF_964205035.1_18MA0850_genomic     plasmid   \n",
       "91823                   GCF_964205045.1_18MA0854_genomic     plasmid   \n",
       "91824                   GCF_964205045.1_18MA0854_genomic     plasmid   \n",
       "\n",
       "                                  SeqID     AAAA     AAAC     AAAG     AAAT  \\\n",
       "0      GCA_000005825.2_ASM582v2_genomic  90881.0  42325.0  60895.0  68917.0   \n",
       "1      GCA_000005845.2_ASM584v2_genomic  70767.0  50477.0  45989.0  51593.0   \n",
       "2      GCA_000006605.1_ASM660v1_genomic  10309.0  11534.0  12081.0   6826.0   \n",
       "3      GCA_000006625.1_ASM662v1_genomic  43692.0  13073.0  11309.0  33335.0   \n",
       "4      GCA_000006645.1_ASM664v1_genomic  75622.0  47896.0  43794.0  60630.0   \n",
       "...                                 ...      ...      ...      ...      ...   \n",
       "91820                     NZ_OZ111549.1     79.0     30.0     32.0     24.0   \n",
       "91821                     NZ_OZ123584.1    131.0    191.0    158.0    106.0   \n",
       "91822                     NZ_OZ123585.1     79.0     86.0     75.0     57.0   \n",
       "91823                     NZ_OZ123587.1    119.0    173.0    133.0    100.0   \n",
       "91824                     NZ_OZ123588.1     79.0     86.0     75.0     57.0   \n",
       "\n",
       "          AACA     AACC     AACG  ...     TAAA     TACA     TAGA     TATA  \\\n",
       "0      43978.0  24963.0  27889.0  ...  67461.0  33931.0  31822.0  17586.0   \n",
       "1      43674.0  40859.0  48632.0  ...  44665.0  21282.0  11307.0   9081.0   \n",
       "2      13481.0  18306.0  17508.0  ...   5623.0   7450.0   7014.0   1039.0   \n",
       "3      13659.0   4502.0   3782.0  ...  33630.0   5964.0   5654.0   5887.0   \n",
       "4      46161.0  43690.0  39865.0  ...  55190.0  23874.0  17241.0  13213.0   \n",
       "...        ...      ...      ...  ...      ...      ...      ...      ...   \n",
       "91820     15.0     13.0     31.0  ...     23.0     18.0     16.0      5.0   \n",
       "91821    243.0    370.0    404.0  ...     36.0    123.0     83.0      7.0   \n",
       "91822     88.0    160.0    143.0  ...     24.0     55.0     30.0      5.0   \n",
       "91823    224.0    315.0    356.0  ...     28.0    110.0     75.0      7.0   \n",
       "91824     88.0    160.0    143.0  ...     24.0     55.0     30.0      5.0   \n",
       "\n",
       "          TCAA     TCCA     TCGA     TGAA     TGCA     TTAA  \n",
       "0      52989.0  30674.0   9695.0  59606.0  17647.0  31864.0  \n",
       "1      38771.0  33770.0  15462.0  51762.0  19768.0  21209.0  \n",
       "2      12658.0  24371.0  12682.0  15320.0   9582.0   2712.0  \n",
       "3      13983.0   3824.0    912.0  12956.0   2571.0  16985.0  \n",
       "4      48290.0  32235.0  11718.0  51982.0  18633.0  23183.0  \n",
       "...        ...      ...      ...      ...      ...      ...  \n",
       "91820     33.0     20.0      6.0     29.0      6.0     10.0  \n",
       "91821    267.0    294.0    364.0    213.0    174.0     15.0  \n",
       "91822    108.0    110.0    144.0     86.0     74.0     12.0  \n",
       "91823    233.0    261.0    326.0    191.0    157.0     13.0  \n",
       "91824    108.0    110.0    144.0     86.0     74.0     12.0  \n",
       "\n",
       "[91825 rows x 139 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load 4mers\n",
    "fn = '/fs/nexus-projects/plasmids/plasmid_host_range_2024_summer_interns/expanded_dataset/4mer_table.txt'\n",
    "data_4mers = pd.read_csv(fn, sep='\\t')\n",
    "data_4mers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chromosome']\n",
      "['plasmid']\n"
     ]
    }
   ],
   "source": [
    "# split 4mers into chromosomes/plasmids\n",
    "c_idx = data_4mers.index[data_4mers['Seq_type'] == 'chromosome']\n",
    "c_4mers = data_4mers.iloc[c_idx]\n",
    "p_4mers = data_4mers.iloc[~c_idx]\n",
    "print(c_4mers['Seq_type'].unique())\n",
    "print(p_4mers['Seq_type'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Seq_type</th>\n",
       "      <th>AAAA</th>\n",
       "      <th>AAAC</th>\n",
       "      <th>AAAG</th>\n",
       "      <th>AAAT</th>\n",
       "      <th>AACA</th>\n",
       "      <th>AACC</th>\n",
       "      <th>AACG</th>\n",
       "      <th>AACT</th>\n",
       "      <th>...</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TACA</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TATA</th>\n",
       "      <th>TCAA</th>\n",
       "      <th>TCCA</th>\n",
       "      <th>TCGA</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGCA</th>\n",
       "      <th>TTAA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SeqID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCA_023703895.1_ASM2370389v1_genomic</th>\n",
       "      <td>GCA_023703895.1_ASM2370389v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>56811.0</td>\n",
       "      <td>39993.0</td>\n",
       "      <td>42497.0</td>\n",
       "      <td>37034.0</td>\n",
       "      <td>33033.0</td>\n",
       "      <td>36321.0</td>\n",
       "      <td>41485.0</td>\n",
       "      <td>22786.0</td>\n",
       "      <td>...</td>\n",
       "      <td>34999.0</td>\n",
       "      <td>16208.0</td>\n",
       "      <td>11026.0</td>\n",
       "      <td>8255.0</td>\n",
       "      <td>32990.0</td>\n",
       "      <td>37421.0</td>\n",
       "      <td>20935.0</td>\n",
       "      <td>42555.0</td>\n",
       "      <td>19699.0</td>\n",
       "      <td>15680.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024662395.1_ASM2466239v1_genomic</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>15936.0</td>\n",
       "      <td>17157.0</td>\n",
       "      <td>13250.0</td>\n",
       "      <td>16569.0</td>\n",
       "      <td>26340.0</td>\n",
       "      <td>26860.0</td>\n",
       "      <td>47610.0</td>\n",
       "      <td>19874.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4780.0</td>\n",
       "      <td>12342.0</td>\n",
       "      <td>10419.0</td>\n",
       "      <td>2043.0</td>\n",
       "      <td>25286.0</td>\n",
       "      <td>32144.0</td>\n",
       "      <td>64817.0</td>\n",
       "      <td>34361.0</td>\n",
       "      <td>23089.0</td>\n",
       "      <td>1488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024662395.1_ASM2466239v1_genomic</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>15936.0</td>\n",
       "      <td>17157.0</td>\n",
       "      <td>13250.0</td>\n",
       "      <td>16569.0</td>\n",
       "      <td>26340.0</td>\n",
       "      <td>26860.0</td>\n",
       "      <td>47610.0</td>\n",
       "      <td>19874.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4780.0</td>\n",
       "      <td>12342.0</td>\n",
       "      <td>10419.0</td>\n",
       "      <td>2043.0</td>\n",
       "      <td>25286.0</td>\n",
       "      <td>32144.0</td>\n",
       "      <td>64817.0</td>\n",
       "      <td>34361.0</td>\n",
       "      <td>23089.0</td>\n",
       "      <td>1488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024662395.1_ASM2466239v1_genomic</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>15936.0</td>\n",
       "      <td>17157.0</td>\n",
       "      <td>13250.0</td>\n",
       "      <td>16569.0</td>\n",
       "      <td>26340.0</td>\n",
       "      <td>26860.0</td>\n",
       "      <td>47610.0</td>\n",
       "      <td>19874.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4780.0</td>\n",
       "      <td>12342.0</td>\n",
       "      <td>10419.0</td>\n",
       "      <td>2043.0</td>\n",
       "      <td>25286.0</td>\n",
       "      <td>32144.0</td>\n",
       "      <td>64817.0</td>\n",
       "      <td>34361.0</td>\n",
       "      <td>23089.0</td>\n",
       "      <td>1488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_012971805.2_ASM1297180v2_genomic</th>\n",
       "      <td>GCA_012971805.2_ASM1297180v2_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>7568.0</td>\n",
       "      <td>20515.0</td>\n",
       "      <td>13250.0</td>\n",
       "      <td>12206.0</td>\n",
       "      <td>40854.0</td>\n",
       "      <td>56397.0</td>\n",
       "      <td>69700.0</td>\n",
       "      <td>41261.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2512.0</td>\n",
       "      <td>20100.0</td>\n",
       "      <td>12931.0</td>\n",
       "      <td>1242.0</td>\n",
       "      <td>31294.0</td>\n",
       "      <td>62458.0</td>\n",
       "      <td>102586.0</td>\n",
       "      <td>40596.0</td>\n",
       "      <td>30106.0</td>\n",
       "      <td>899.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003595385.1_ASM359538v1_genomic</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>88416.0</td>\n",
       "      <td>36930.0</td>\n",
       "      <td>44999.0</td>\n",
       "      <td>85338.0</td>\n",
       "      <td>44229.0</td>\n",
       "      <td>16772.0</td>\n",
       "      <td>19766.0</td>\n",
       "      <td>30342.0</td>\n",
       "      <td>...</td>\n",
       "      <td>80969.0</td>\n",
       "      <td>31753.0</td>\n",
       "      <td>25172.0</td>\n",
       "      <td>23405.0</td>\n",
       "      <td>46090.0</td>\n",
       "      <td>18894.0</td>\n",
       "      <td>7004.0</td>\n",
       "      <td>48760.0</td>\n",
       "      <td>14197.0</td>\n",
       "      <td>37970.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003595385.1_ASM359538v1_genomic</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>88416.0</td>\n",
       "      <td>36930.0</td>\n",
       "      <td>44999.0</td>\n",
       "      <td>85338.0</td>\n",
       "      <td>44229.0</td>\n",
       "      <td>16772.0</td>\n",
       "      <td>19766.0</td>\n",
       "      <td>30342.0</td>\n",
       "      <td>...</td>\n",
       "      <td>80969.0</td>\n",
       "      <td>31753.0</td>\n",
       "      <td>25172.0</td>\n",
       "      <td>23405.0</td>\n",
       "      <td>46090.0</td>\n",
       "      <td>18894.0</td>\n",
       "      <td>7004.0</td>\n",
       "      <td>48760.0</td>\n",
       "      <td>14197.0</td>\n",
       "      <td>37970.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_039640675.1_ASM3964067v1_genomic</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>67177.0</td>\n",
       "      <td>30928.0</td>\n",
       "      <td>39504.0</td>\n",
       "      <td>66957.0</td>\n",
       "      <td>32348.0</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>16162.0</td>\n",
       "      <td>27851.0</td>\n",
       "      <td>...</td>\n",
       "      <td>69879.0</td>\n",
       "      <td>24925.0</td>\n",
       "      <td>24263.0</td>\n",
       "      <td>19210.0</td>\n",
       "      <td>38717.0</td>\n",
       "      <td>16109.0</td>\n",
       "      <td>5298.0</td>\n",
       "      <td>42747.0</td>\n",
       "      <td>9307.0</td>\n",
       "      <td>32466.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_039640675.1_ASM3964067v1_genomic</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>67177.0</td>\n",
       "      <td>30928.0</td>\n",
       "      <td>39504.0</td>\n",
       "      <td>66957.0</td>\n",
       "      <td>32348.0</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>16162.0</td>\n",
       "      <td>27851.0</td>\n",
       "      <td>...</td>\n",
       "      <td>69879.0</td>\n",
       "      <td>24925.0</td>\n",
       "      <td>24263.0</td>\n",
       "      <td>19210.0</td>\n",
       "      <td>38717.0</td>\n",
       "      <td>16109.0</td>\n",
       "      <td>5298.0</td>\n",
       "      <td>42747.0</td>\n",
       "      <td>9307.0</td>\n",
       "      <td>32466.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_039640675.1_ASM3964067v1_genomic</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>67177.0</td>\n",
       "      <td>30928.0</td>\n",
       "      <td>39504.0</td>\n",
       "      <td>66957.0</td>\n",
       "      <td>32348.0</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>16162.0</td>\n",
       "      <td>27851.0</td>\n",
       "      <td>...</td>\n",
       "      <td>69879.0</td>\n",
       "      <td>24925.0</td>\n",
       "      <td>24263.0</td>\n",
       "      <td>19210.0</td>\n",
       "      <td>38717.0</td>\n",
       "      <td>16109.0</td>\n",
       "      <td>5298.0</td>\n",
       "      <td>42747.0</td>\n",
       "      <td>9307.0</td>\n",
       "      <td>32466.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50851 rows Ã— 138 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                  Assembly  \\\n",
       "SeqID                                                                        \n",
       "GCA_023703895.1_ASM2370389v1_genomic  GCA_023703895.1_ASM2370389v1_genomic   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  GCF_024662395.1_ASM2466239v1_genomic   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  GCF_024662395.1_ASM2466239v1_genomic   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  GCF_024662395.1_ASM2466239v1_genomic   \n",
       "GCA_012971805.2_ASM1297180v2_genomic  GCA_012971805.2_ASM1297180v2_genomic   \n",
       "...                                                                    ...   \n",
       "GCF_003595385.1_ASM359538v1_genomic    GCF_003595385.1_ASM359538v1_genomic   \n",
       "GCF_003595385.1_ASM359538v1_genomic    GCF_003595385.1_ASM359538v1_genomic   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  GCF_039640675.1_ASM3964067v1_genomic   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  GCF_039640675.1_ASM3964067v1_genomic   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  GCF_039640675.1_ASM3964067v1_genomic   \n",
       "\n",
       "                                        Seq_type     AAAA     AAAC     AAAG  \\\n",
       "SeqID                                                                         \n",
       "GCA_023703895.1_ASM2370389v1_genomic  chromosome  56811.0  39993.0  42497.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  chromosome  15936.0  17157.0  13250.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  chromosome  15936.0  17157.0  13250.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  chromosome  15936.0  17157.0  13250.0   \n",
       "GCA_012971805.2_ASM1297180v2_genomic  chromosome   7568.0  20515.0  13250.0   \n",
       "...                                          ...      ...      ...      ...   \n",
       "GCF_003595385.1_ASM359538v1_genomic   chromosome  88416.0  36930.0  44999.0   \n",
       "GCF_003595385.1_ASM359538v1_genomic   chromosome  88416.0  36930.0  44999.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  chromosome  67177.0  30928.0  39504.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  chromosome  67177.0  30928.0  39504.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  chromosome  67177.0  30928.0  39504.0   \n",
       "\n",
       "                                         AAAT     AACA     AACC     AACG  \\\n",
       "SeqID                                                                      \n",
       "GCA_023703895.1_ASM2370389v1_genomic  37034.0  33033.0  36321.0  41485.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  16569.0  26340.0  26860.0  47610.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  16569.0  26340.0  26860.0  47610.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  16569.0  26340.0  26860.0  47610.0   \n",
       "GCA_012971805.2_ASM1297180v2_genomic  12206.0  40854.0  56397.0  69700.0   \n",
       "...                                       ...      ...      ...      ...   \n",
       "GCF_003595385.1_ASM359538v1_genomic   85338.0  44229.0  16772.0  19766.0   \n",
       "GCF_003595385.1_ASM359538v1_genomic   85338.0  44229.0  16772.0  19766.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  66957.0  32348.0  14958.0  16162.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  66957.0  32348.0  14958.0  16162.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  66957.0  32348.0  14958.0  16162.0   \n",
       "\n",
       "                                         AACT  ...     TAAA     TACA     TAGA  \\\n",
       "SeqID                                          ...                              \n",
       "GCA_023703895.1_ASM2370389v1_genomic  22786.0  ...  34999.0  16208.0  11026.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  19874.0  ...   4780.0  12342.0  10419.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  19874.0  ...   4780.0  12342.0  10419.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic  19874.0  ...   4780.0  12342.0  10419.0   \n",
       "GCA_012971805.2_ASM1297180v2_genomic  41261.0  ...   2512.0  20100.0  12931.0   \n",
       "...                                       ...  ...      ...      ...      ...   \n",
       "GCF_003595385.1_ASM359538v1_genomic   30342.0  ...  80969.0  31753.0  25172.0   \n",
       "GCF_003595385.1_ASM359538v1_genomic   30342.0  ...  80969.0  31753.0  25172.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  27851.0  ...  69879.0  24925.0  24263.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  27851.0  ...  69879.0  24925.0  24263.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  27851.0  ...  69879.0  24925.0  24263.0   \n",
       "\n",
       "                                         TATA     TCAA     TCCA      TCGA  \\\n",
       "SeqID                                                                       \n",
       "GCA_023703895.1_ASM2370389v1_genomic   8255.0  32990.0  37421.0   20935.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic   2043.0  25286.0  32144.0   64817.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic   2043.0  25286.0  32144.0   64817.0   \n",
       "GCF_024662395.1_ASM2466239v1_genomic   2043.0  25286.0  32144.0   64817.0   \n",
       "GCA_012971805.2_ASM1297180v2_genomic   1242.0  31294.0  62458.0  102586.0   \n",
       "...                                       ...      ...      ...       ...   \n",
       "GCF_003595385.1_ASM359538v1_genomic   23405.0  46090.0  18894.0    7004.0   \n",
       "GCF_003595385.1_ASM359538v1_genomic   23405.0  46090.0  18894.0    7004.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  19210.0  38717.0  16109.0    5298.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  19210.0  38717.0  16109.0    5298.0   \n",
       "GCF_039640675.1_ASM3964067v1_genomic  19210.0  38717.0  16109.0    5298.0   \n",
       "\n",
       "                                         TGAA     TGCA     TTAA  \n",
       "SeqID                                                            \n",
       "GCA_023703895.1_ASM2370389v1_genomic  42555.0  19699.0  15680.0  \n",
       "GCF_024662395.1_ASM2466239v1_genomic  34361.0  23089.0   1488.0  \n",
       "GCF_024662395.1_ASM2466239v1_genomic  34361.0  23089.0   1488.0  \n",
       "GCF_024662395.1_ASM2466239v1_genomic  34361.0  23089.0   1488.0  \n",
       "GCA_012971805.2_ASM1297180v2_genomic  40596.0  30106.0    899.0  \n",
       "...                                       ...      ...      ...  \n",
       "GCF_003595385.1_ASM359538v1_genomic   48760.0  14197.0  37970.0  \n",
       "GCF_003595385.1_ASM359538v1_genomic   48760.0  14197.0  37970.0  \n",
       "GCF_039640675.1_ASM3964067v1_genomic  42747.0   9307.0  32466.0  \n",
       "GCF_039640675.1_ASM3964067v1_genomic  42747.0   9307.0  32466.0  \n",
       "GCF_039640675.1_ASM3964067v1_genomic  42747.0   9307.0  32466.0  \n",
       "\n",
       "[50851 rows x 138 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get chromosome 4mers in order of pairwise dataset\n",
    "labels = data['Assembly_ID'].str[:-7] # strip filename\n",
    "_c4 = c_4mers.set_index('SeqID')\n",
    "# a few labels are not found in c_4mers, so just drop them i guess\n",
    "labels_to_keep = labels.isin(c_4mers['SeqID'])\n",
    "labels = labels[labels_to_keep]\n",
    "\n",
    "c4m = _c4.loc[labels]\n",
    "c4m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Seq_type</th>\n",
       "      <th>AAAA</th>\n",
       "      <th>AAAC</th>\n",
       "      <th>AAAG</th>\n",
       "      <th>AAAT</th>\n",
       "      <th>AACA</th>\n",
       "      <th>AACC</th>\n",
       "      <th>AACG</th>\n",
       "      <th>AACT</th>\n",
       "      <th>...</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TACA</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TATA</th>\n",
       "      <th>TCAA</th>\n",
       "      <th>TCCA</th>\n",
       "      <th>TCGA</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGCA</th>\n",
       "      <th>TTAA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SeqID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CP077813.1</th>\n",
       "      <td>GCA_023703895.1_ASM2370389v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>824.0</td>\n",
       "      <td>547.0</td>\n",
       "      <td>521.0</td>\n",
       "      <td>559.0</td>\n",
       "      <td>446.0</td>\n",
       "      <td>491.0</td>\n",
       "      <td>501.0</td>\n",
       "      <td>349.0</td>\n",
       "      <td>...</td>\n",
       "      <td>468.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>418.0</td>\n",
       "      <td>426.0</td>\n",
       "      <td>211.0</td>\n",
       "      <td>558.0</td>\n",
       "      <td>209.0</td>\n",
       "      <td>230.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP100133.1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>684.0</td>\n",
       "      <td>763.0</td>\n",
       "      <td>635.0</td>\n",
       "      <td>635.0</td>\n",
       "      <td>838.0</td>\n",
       "      <td>1019.0</td>\n",
       "      <td>1727.0</td>\n",
       "      <td>809.0</td>\n",
       "      <td>...</td>\n",
       "      <td>230.0</td>\n",
       "      <td>453.0</td>\n",
       "      <td>335.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>1139.0</td>\n",
       "      <td>1158.0</td>\n",
       "      <td>1726.0</td>\n",
       "      <td>1134.0</td>\n",
       "      <td>700.0</td>\n",
       "      <td>101.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP100134.1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>581.0</td>\n",
       "      <td>674.0</td>\n",
       "      <td>506.0</td>\n",
       "      <td>594.0</td>\n",
       "      <td>683.0</td>\n",
       "      <td>713.0</td>\n",
       "      <td>1188.0</td>\n",
       "      <td>599.0</td>\n",
       "      <td>...</td>\n",
       "      <td>220.0</td>\n",
       "      <td>348.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>832.0</td>\n",
       "      <td>879.0</td>\n",
       "      <td>1127.0</td>\n",
       "      <td>864.0</td>\n",
       "      <td>514.0</td>\n",
       "      <td>85.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP100135.1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>421.0</td>\n",
       "      <td>405.0</td>\n",
       "      <td>403.0</td>\n",
       "      <td>386.0</td>\n",
       "      <td>448.0</td>\n",
       "      <td>561.0</td>\n",
       "      <td>894.0</td>\n",
       "      <td>431.0</td>\n",
       "      <td>...</td>\n",
       "      <td>142.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>654.0</td>\n",
       "      <td>626.0</td>\n",
       "      <td>932.0</td>\n",
       "      <td>598.0</td>\n",
       "      <td>379.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP115646.1</th>\n",
       "      <td>GCF_024106315.1_ASM2410631v2_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>2197.0</td>\n",
       "      <td>1453.0</td>\n",
       "      <td>1528.0</td>\n",
       "      <td>1804.0</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>854.0</td>\n",
       "      <td>954.0</td>\n",
       "      <td>1121.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1584.0</td>\n",
       "      <td>732.0</td>\n",
       "      <td>646.0</td>\n",
       "      <td>270.0</td>\n",
       "      <td>1812.0</td>\n",
       "      <td>745.0</td>\n",
       "      <td>405.0</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>452.0</td>\n",
       "      <td>822.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP014366.1</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>223.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>...</td>\n",
       "      <td>119.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP014367.1</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>1170.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>505.0</td>\n",
       "      <td>876.0</td>\n",
       "      <td>340.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>285.0</td>\n",
       "      <td>...</td>\n",
       "      <td>896.0</td>\n",
       "      <td>282.0</td>\n",
       "      <td>267.0</td>\n",
       "      <td>315.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>415.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>373.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP098817.1</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>1872.0</td>\n",
       "      <td>601.0</td>\n",
       "      <td>783.0</td>\n",
       "      <td>1408.0</td>\n",
       "      <td>620.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>269.0</td>\n",
       "      <td>527.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1425.0</td>\n",
       "      <td>515.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>499.0</td>\n",
       "      <td>636.0</td>\n",
       "      <td>252.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>635.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>595.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP098818.1</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>1725.0</td>\n",
       "      <td>578.0</td>\n",
       "      <td>868.0</td>\n",
       "      <td>1258.0</td>\n",
       "      <td>615.0</td>\n",
       "      <td>207.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>452.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1321.0</td>\n",
       "      <td>420.0</td>\n",
       "      <td>487.0</td>\n",
       "      <td>488.0</td>\n",
       "      <td>607.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>676.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>536.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NZ_CP098819.1</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic</td>\n",
       "      <td>plasmid</td>\n",
       "      <td>1068.0</td>\n",
       "      <td>291.0</td>\n",
       "      <td>443.0</td>\n",
       "      <td>741.0</td>\n",
       "      <td>307.0</td>\n",
       "      <td>132.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>...</td>\n",
       "      <td>758.0</td>\n",
       "      <td>262.0</td>\n",
       "      <td>237.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>327.0</td>\n",
       "      <td>132.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>366.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>337.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42637 rows Ã— 138 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Assembly Seq_type    AAAA    AAAC  \\\n",
       "SeqID                                                                          \n",
       "CP077813.1     GCA_023703895.1_ASM2370389v1_genomic  plasmid   824.0   547.0   \n",
       "NZ_CP100133.1  GCF_024662395.1_ASM2466239v1_genomic  plasmid   684.0   763.0   \n",
       "NZ_CP100134.1  GCF_024662395.1_ASM2466239v1_genomic  plasmid   581.0   674.0   \n",
       "NZ_CP100135.1  GCF_024662395.1_ASM2466239v1_genomic  plasmid   421.0   405.0   \n",
       "NZ_CP115646.1  GCF_024106315.1_ASM2410631v2_genomic  plasmid  2197.0  1453.0   \n",
       "...                                             ...      ...     ...     ...   \n",
       "NZ_CP014366.1   GCF_003595385.1_ASM359538v1_genomic  plasmid   223.0    69.0   \n",
       "NZ_CP014367.1   GCF_003595385.1_ASM359538v1_genomic  plasmid  1170.0   404.0   \n",
       "NZ_CP098817.1  GCF_039640675.1_ASM3964067v1_genomic  plasmid  1872.0   601.0   \n",
       "NZ_CP098818.1  GCF_039640675.1_ASM3964067v1_genomic  plasmid  1725.0   578.0   \n",
       "NZ_CP098819.1  GCF_039640675.1_ASM3964067v1_genomic  plasmid  1068.0   291.0   \n",
       "\n",
       "                 AAAG    AAAT    AACA    AACC    AACG    AACT  ...    TAAA  \\\n",
       "SeqID                                                          ...           \n",
       "CP077813.1      521.0   559.0   446.0   491.0   501.0   349.0  ...   468.0   \n",
       "NZ_CP100133.1   635.0   635.0   838.0  1019.0  1727.0   809.0  ...   230.0   \n",
       "NZ_CP100134.1   506.0   594.0   683.0   713.0  1188.0   599.0  ...   220.0   \n",
       "NZ_CP100135.1   403.0   386.0   448.0   561.0   894.0   431.0  ...   142.0   \n",
       "NZ_CP115646.1  1528.0  1804.0  1287.0   854.0   954.0  1121.0  ...  1584.0   \n",
       "...               ...     ...     ...     ...     ...     ...  ...     ...   \n",
       "NZ_CP014366.1    62.0   113.0    55.0    27.0    20.0    42.0  ...   119.0   \n",
       "NZ_CP014367.1   505.0   876.0   340.0   170.0   166.0   285.0  ...   896.0   \n",
       "NZ_CP098817.1   783.0  1408.0   620.0   238.0   269.0   527.0  ...  1425.0   \n",
       "NZ_CP098818.1   868.0  1258.0   615.0   207.0   200.0   452.0  ...  1321.0   \n",
       "NZ_CP098819.1   443.0   741.0   307.0   132.0   124.0   245.0  ...   758.0   \n",
       "\n",
       "                TACA   TAGA   TATA    TCAA    TCCA    TCGA    TGAA   TGCA  \\\n",
       "SeqID                                                                       \n",
       "CP077813.1     202.0  117.0  123.0   418.0   426.0   211.0   558.0  209.0   \n",
       "NZ_CP100133.1  453.0  335.0   86.0  1139.0  1158.0  1726.0  1134.0  700.0   \n",
       "NZ_CP100134.1  348.0  255.0   85.0   832.0   879.0  1127.0   864.0  514.0   \n",
       "NZ_CP100135.1  243.0  219.0   61.0   654.0   626.0   932.0   598.0  379.0   \n",
       "NZ_CP115646.1  732.0  646.0  270.0  1812.0   745.0   405.0  1537.0  452.0   \n",
       "...              ...    ...    ...     ...     ...     ...     ...    ...   \n",
       "NZ_CP014366.1   23.0   28.0   31.0    54.0    18.0    11.0    40.0   10.0   \n",
       "NZ_CP014367.1  282.0  267.0  315.0   394.0   167.0    65.0   415.0   98.0   \n",
       "NZ_CP098817.1  515.0  427.0  499.0   636.0   252.0    84.0   635.0  139.0   \n",
       "NZ_CP098818.1  420.0  487.0  488.0   607.0   202.0    47.0   676.0   69.0   \n",
       "NZ_CP098819.1  262.0  237.0  300.0   327.0   132.0    44.0   366.0   73.0   \n",
       "\n",
       "                TTAA  \n",
       "SeqID                 \n",
       "CP077813.1     230.0  \n",
       "NZ_CP100133.1  101.0  \n",
       "NZ_CP100134.1   85.0  \n",
       "NZ_CP100135.1   50.0  \n",
       "NZ_CP115646.1  822.0  \n",
       "...              ...  \n",
       "NZ_CP014366.1   56.0  \n",
       "NZ_CP014367.1  373.0  \n",
       "NZ_CP098817.1  595.0  \n",
       "NZ_CP098818.1  536.0  \n",
       "NZ_CP098819.1  337.0  \n",
       "\n",
       "[42637 rows x 138 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels = data['Plasmid_ID']\n",
    "# drop labels\n",
    "labels = labels[labels_to_keep]\n",
    "_p4 = p_4mers.set_index('SeqID')\n",
    "\n",
    "labels_to_keep_p = labels.isin(p_4mers['SeqID'])\n",
    "labels = labels[labels_to_keep_p]\n",
    "\n",
    "p4m = _p4.loc[labels]\n",
    "p4m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AAAA</th>\n",
       "      <th>AAAC</th>\n",
       "      <th>AAAG</th>\n",
       "      <th>AAAT</th>\n",
       "      <th>AACA</th>\n",
       "      <th>AACC</th>\n",
       "      <th>AACG</th>\n",
       "      <th>AACT</th>\n",
       "      <th>AAGA</th>\n",
       "      <th>AAGC</th>\n",
       "      <th>...</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TACA</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TATA</th>\n",
       "      <th>TCAA</th>\n",
       "      <th>TCCA</th>\n",
       "      <th>TCGA</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGCA</th>\n",
       "      <th>TTAA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.004185</td>\n",
       "      <td>0.002337</td>\n",
       "      <td>0.001374</td>\n",
       "      <td>0.003131</td>\n",
       "      <td>0.001823</td>\n",
       "      <td>0.002016</td>\n",
       "      <td>0.001201</td>\n",
       "      <td>0.002020</td>\n",
       "      <td>-0.000110</td>\n",
       "      <td>-0.000266</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001848</td>\n",
       "      <td>0.000585</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000668</td>\n",
       "      <td>0.001316</td>\n",
       "      <td>0.000606</td>\n",
       "      <td>-0.000164</td>\n",
       "      <td>0.002044</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.001203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.001183</td>\n",
       "      <td>0.001414</td>\n",
       "      <td>0.001334</td>\n",
       "      <td>0.000828</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.001287</td>\n",
       "      <td>0.001862</td>\n",
       "      <td>0.001243</td>\n",
       "      <td>0.001043</td>\n",
       "      <td>0.001101</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000486</td>\n",
       "      <td>0.000511</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.002160</td>\n",
       "      <td>0.001215</td>\n",
       "      <td>-0.000766</td>\n",
       "      <td>0.000750</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.002077</td>\n",
       "      <td>0.002613</td>\n",
       "      <td>0.001905</td>\n",
       "      <td>0.002082</td>\n",
       "      <td>0.001282</td>\n",
       "      <td>0.001436</td>\n",
       "      <td>0.001957</td>\n",
       "      <td>0.001617</td>\n",
       "      <td>0.001071</td>\n",
       "      <td>0.001933</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000978</td>\n",
       "      <td>0.000818</td>\n",
       "      <td>0.000390</td>\n",
       "      <td>0.000348</td>\n",
       "      <td>0.002599</td>\n",
       "      <td>0.001918</td>\n",
       "      <td>-0.001140</td>\n",
       "      <td>0.001464</td>\n",
       "      <td>0.000467</td>\n",
       "      <td>0.000433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.001590</td>\n",
       "      <td>0.001251</td>\n",
       "      <td>0.001828</td>\n",
       "      <td>0.001159</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.001261</td>\n",
       "      <td>0.001277</td>\n",
       "      <td>0.001085</td>\n",
       "      <td>0.001328</td>\n",
       "      <td>0.001408</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000627</td>\n",
       "      <td>0.000438</td>\n",
       "      <td>0.000502</td>\n",
       "      <td>0.000271</td>\n",
       "      <td>0.002389</td>\n",
       "      <td>0.001076</td>\n",
       "      <td>-0.000984</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>-0.007263</td>\n",
       "      <td>-0.000025</td>\n",
       "      <td>-0.000632</td>\n",
       "      <td>-0.004878</td>\n",
       "      <td>0.000528</td>\n",
       "      <td>-0.000807</td>\n",
       "      <td>0.002573</td>\n",
       "      <td>-0.000400</td>\n",
       "      <td>0.001311</td>\n",
       "      <td>0.000960</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.008027</td>\n",
       "      <td>-0.001340</td>\n",
       "      <td>-0.000483</td>\n",
       "      <td>-0.001589</td>\n",
       "      <td>0.003062</td>\n",
       "      <td>0.000417</td>\n",
       "      <td>0.001412</td>\n",
       "      <td>-0.000054</td>\n",
       "      <td>-0.001719</td>\n",
       "      <td>-0.001950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50871</th>\n",
       "      <td>0.040992</td>\n",
       "      <td>0.009388</td>\n",
       "      <td>0.004369</td>\n",
       "      <td>0.006818</td>\n",
       "      <td>0.002392</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>-0.000398</td>\n",
       "      <td>0.003008</td>\n",
       "      <td>0.003382</td>\n",
       "      <td>-0.000987</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010244</td>\n",
       "      <td>-0.003564</td>\n",
       "      <td>0.000303</td>\n",
       "      <td>0.001873</td>\n",
       "      <td>0.001431</td>\n",
       "      <td>-0.000739</td>\n",
       "      <td>0.001112</td>\n",
       "      <td>-0.003973</td>\n",
       "      <td>-0.001684</td>\n",
       "      <td>0.004866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50872</th>\n",
       "      <td>0.012793</td>\n",
       "      <td>0.002214</td>\n",
       "      <td>0.003168</td>\n",
       "      <td>0.002990</td>\n",
       "      <td>-0.002663</td>\n",
       "      <td>0.000508</td>\n",
       "      <td>-0.000671</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>0.002121</td>\n",
       "      <td>-0.000164</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005233</td>\n",
       "      <td>-0.000511</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.003582</td>\n",
       "      <td>-0.001308</td>\n",
       "      <td>-0.000334</td>\n",
       "      <td>-0.000009</td>\n",
       "      <td>-0.001452</td>\n",
       "      <td>-0.001266</td>\n",
       "      <td>0.000711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50873</th>\n",
       "      <td>0.014920</td>\n",
       "      <td>0.000959</td>\n",
       "      <td>0.001572</td>\n",
       "      <td>0.004497</td>\n",
       "      <td>0.000808</td>\n",
       "      <td>-0.000730</td>\n",
       "      <td>-0.000520</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>0.000639</td>\n",
       "      <td>-0.001268</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003686</td>\n",
       "      <td>0.001467</td>\n",
       "      <td>-0.000256</td>\n",
       "      <td>0.003444</td>\n",
       "      <td>-0.001436</td>\n",
       "      <td>-0.000884</td>\n",
       "      <td>-0.000265</td>\n",
       "      <td>-0.003109</td>\n",
       "      <td>-0.000660</td>\n",
       "      <td>0.000193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50874</th>\n",
       "      <td>0.016174</td>\n",
       "      <td>0.001974</td>\n",
       "      <td>0.005805</td>\n",
       "      <td>0.004442</td>\n",
       "      <td>0.002329</td>\n",
       "      <td>-0.000882</td>\n",
       "      <td>-0.001552</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.004587</td>\n",
       "      <td>-0.001617</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004841</td>\n",
       "      <td>0.000431</td>\n",
       "      <td>0.002398</td>\n",
       "      <td>0.004492</td>\n",
       "      <td>-0.000480</td>\n",
       "      <td>-0.001480</td>\n",
       "      <td>-0.000979</td>\n",
       "      <td>-0.000383</td>\n",
       "      <td>-0.002063</td>\n",
       "      <td>0.000281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50875</th>\n",
       "      <td>0.019812</td>\n",
       "      <td>0.000231</td>\n",
       "      <td>0.003454</td>\n",
       "      <td>0.005417</td>\n",
       "      <td>0.000359</td>\n",
       "      <td>-0.000275</td>\n",
       "      <td>-0.001123</td>\n",
       "      <td>-0.000547</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>-0.001530</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004974</td>\n",
       "      <td>0.001404</td>\n",
       "      <td>0.000567</td>\n",
       "      <td>0.005426</td>\n",
       "      <td>-0.001362</td>\n",
       "      <td>-0.000746</td>\n",
       "      <td>-0.000219</td>\n",
       "      <td>-0.001284</td>\n",
       "      <td>-0.000576</td>\n",
       "      <td>0.001639</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42637 rows Ã— 136 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           AAAA      AAAC      AAAG      AAAT      AACA      AACC      AACG  \\\n",
       "Index                                                                         \n",
       "0      0.004185  0.002337  0.001374  0.003131  0.001823  0.002016  0.001201   \n",
       "1      0.001183  0.001414  0.001334  0.000828  0.000410  0.001287  0.001862   \n",
       "2      0.002077  0.002613  0.001905  0.002082  0.001282  0.001436  0.001957   \n",
       "3      0.001590  0.001251  0.001828  0.001159  0.000261  0.001261  0.001277   \n",
       "6     -0.007263 -0.000025 -0.000632 -0.004878  0.000528 -0.000807  0.002573   \n",
       "...         ...       ...       ...       ...       ...       ...       ...   \n",
       "50871  0.040992  0.009388  0.004369  0.006818  0.002392  0.002875 -0.000398   \n",
       "50872  0.012793  0.002214  0.003168  0.002990 -0.002663  0.000508 -0.000671   \n",
       "50873  0.014920  0.000959  0.001572  0.004497  0.000808 -0.000730 -0.000520   \n",
       "50874  0.016174  0.001974  0.005805  0.004442  0.002329 -0.000882 -0.001552   \n",
       "50875  0.019812  0.000231  0.003454  0.005417  0.000359 -0.000275 -0.001123   \n",
       "\n",
       "           AACT      AAGA      AAGC  ...      TAAA      TACA      TAGA  \\\n",
       "Index                                ...                                 \n",
       "0      0.002020 -0.000110 -0.000266  ...  0.001848  0.000585  0.000022   \n",
       "1      0.001243  0.001043  0.001101  ...  0.000486  0.000511  0.000181   \n",
       "2      0.001617  0.001071  0.001933  ...  0.000978  0.000818  0.000390   \n",
       "3      0.001085  0.001328  0.001408  ...  0.000627  0.000438  0.000502   \n",
       "6     -0.000400  0.001311  0.000960  ... -0.008027 -0.001340 -0.000483   \n",
       "...         ...       ...       ...  ...       ...       ...       ...   \n",
       "50871  0.003008  0.003382 -0.000987  ...  0.010244 -0.003564  0.000303   \n",
       "50872  0.000085  0.002121 -0.000164  ...  0.005233 -0.000511  0.001200   \n",
       "50873  0.000541  0.000639 -0.001268  ...  0.003686  0.001467 -0.000256   \n",
       "50874  0.000043  0.004587 -0.001617  ...  0.004841  0.000431  0.002398   \n",
       "50875 -0.000547  0.000312 -0.001530  ...  0.004974  0.001404  0.000567   \n",
       "\n",
       "           TATA      TCAA      TCCA      TCGA      TGAA      TGCA      TTAA  \n",
       "Index                                                                        \n",
       "0      0.000668  0.001316  0.000606 -0.000164  0.002044  0.000038  0.001203  \n",
       "1      0.000143  0.002160  0.001215 -0.000766  0.000750  0.000177  0.000307  \n",
       "2      0.000348  0.002599  0.001918 -0.001140  0.001464  0.000467  0.000433  \n",
       "3      0.000271  0.002389  0.001076 -0.000984  0.000471  0.000098  0.000251  \n",
       "6     -0.001589  0.003062  0.000417  0.001412 -0.000054 -0.001719 -0.001950  \n",
       "...         ...       ...       ...       ...       ...       ...       ...  \n",
       "50871  0.001873  0.001431 -0.000739  0.001112 -0.003973 -0.001684  0.004866  \n",
       "50872  0.003582 -0.001308 -0.000334 -0.000009 -0.001452 -0.001266  0.000711  \n",
       "50873  0.003444 -0.001436 -0.000884 -0.000265 -0.003109 -0.000660  0.000193  \n",
       "50874  0.004492 -0.000480 -0.001480 -0.000979 -0.000383 -0.002063  0.000281  \n",
       "50875  0.005426 -0.001362 -0.000746 -0.000219 -0.001284 -0.000576  0.001639  \n",
       "\n",
       "[42637 rows x 136 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert to kmer frequencies\n",
    "# ignore terrible variable naming conventions idk what i'm doing here\n",
    "data_c = data[labels_to_keep]\n",
    "data_p = data_c[labels_to_keep_p]\n",
    "p4m_swap = p4m.copy()\n",
    "p4m_swap['SeqID'] = p4m.index\n",
    "p4m_swap['Index'] = data_p.index\n",
    "p4m_swap = p4m_swap.set_index('Index')\n",
    "p_kmers = p4m_swap.iloc[:,2:-1]\n",
    "p_sums = p_kmers.sum(axis=1)\n",
    "p_freqs = p_kmers.div(p_sums, axis=0)\n",
    "\n",
    "c4m['Index'] = data_c.index\n",
    "c4m_red = c4m.set_index('Index')[labels_to_keep_p]\n",
    "c_kmers = c4m_red.iloc[:,2:]\n",
    "c_sums = c_kmers.sum(axis=1)\n",
    "c_freqs = c_kmers.div(c_sums, axis=0)\n",
    "\n",
    "# compute differences\n",
    "diffs = p_freqs.add(c_freqs * -1)\n",
    "diffs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0022763465391264304 +/- 0.0012708710726392703\n"
     ]
    }
   ],
   "source": [
    "# convert to RMSD across all kmers\n",
    "rmsd = diffs ** 2\n",
    "rmsd = rmsd.sum(axis=1)\n",
    "rmsd = rmsd / 136 # * 1/n\n",
    "rmsd = rmsd ** 0.5\n",
    "# mean + std of RMSD across all samples\n",
    "print(rmsd.mean(), '+/-', rmsd.std())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly_ID</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Chromosome_ID</th>\n",
       "      <th>Plasmid_ID</th>\n",
       "      <th>Chr_len</th>\n",
       "      <th>Plasmid_len</th>\n",
       "      <th>Chr_GC</th>\n",
       "      <th>Plasmid_GC</th>\n",
       "      <th>RMSD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_023703895.1_ASM2370389v1_genomic.fna.gz</td>\n",
       "      <td>KLEBSIELLA PNEUMONIAE</td>\n",
       "      <td>CP077812.1</td>\n",
       "      <td>CP077813.1</td>\n",
       "      <td>5167508</td>\n",
       "      <td>54288</td>\n",
       "      <td>0.575169</td>\n",
       "      <td>0.527888</td>\n",
       "      <td>0.001915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100133.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>189346</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.638836</td>\n",
       "      <td>0.002416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100134.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>128913</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.614430</td>\n",
       "      <td>0.003242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100135.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>104740</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.636500</td>\n",
       "      <td>0.002477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_024106315.1_ASM2410631v2_genomic.fna.gz</td>\n",
       "      <td>ACINETOBACTER BAUMANNII</td>\n",
       "      <td>NZ_CP115645.1</td>\n",
       "      <td>NZ_CP115646.1</td>\n",
       "      <td>3917468</td>\n",
       "      <td>104651</td>\n",
       "      <td>0.392244</td>\n",
       "      <td>0.425471</td>\n",
       "      <td>0.001895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50871</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014366.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>3125</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.287360</td>\n",
       "      <td>0.004323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50872</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014367.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>27068</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.305231</td>\n",
       "      <td>0.001853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50873</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098817.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>44138</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.300580</td>\n",
       "      <td>0.001924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50874</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098818.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>39504</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289464</td>\n",
       "      <td>0.002272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50875</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098819.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>22579</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289473</td>\n",
       "      <td>0.002463</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42637 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Assembly_ID                 Taxonomy  \\\n",
       "0      GCA_023703895.1_ASM2370389v1_genomic.fna.gz    KLEBSIELLA PNEUMONIAE   \n",
       "1      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "2      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "3      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "6      GCF_024106315.1_ASM2410631v2_genomic.fna.gz  ACINETOBACTER BAUMANNII   \n",
       "...                                            ...                      ...   \n",
       "50871   GCF_003595385.1_ASM359538v1_genomic.fna.gz    STAPHYLOCOCCUS AUREUS   \n",
       "50872   GCF_003595385.1_ASM359538v1_genomic.fna.gz    STAPHYLOCOCCUS AUREUS   \n",
       "50873  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "50874  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "50875  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "\n",
       "                     Chromosome_ID     Plasmid_ID  Chr_len  Plasmid_len  \\\n",
       "0                       CP077812.1     CP077813.1  5167508        54288   \n",
       "1      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100133.1  6558951       189346   \n",
       "2      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100134.1  6558951       128913   \n",
       "3      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100135.1  6558951       104740   \n",
       "6                    NZ_CP115645.1  NZ_CP115646.1  3917468       104651   \n",
       "...                            ...            ...      ...          ...   \n",
       "50871                NZ_CP014365.1  NZ_CP014366.1  2904958         3125   \n",
       "50872                NZ_CP014365.1  NZ_CP014367.1  2904958        27068   \n",
       "50873                NZ_CP098816.1  NZ_CP098817.1  2443214        44138   \n",
       "50874                NZ_CP098816.1  NZ_CP098818.1  2443214        39504   \n",
       "50875                NZ_CP098816.1  NZ_CP098819.1  2443214        22579   \n",
       "\n",
       "         Chr_GC  Plasmid_GC      RMSD  \n",
       "0      0.575169    0.527888  0.001915  \n",
       "1      0.683257    0.638836  0.002416  \n",
       "2      0.683257    0.614430  0.003242  \n",
       "3      0.683257    0.636500  0.002477  \n",
       "6      0.392244    0.425471  0.001895  \n",
       "...         ...         ...       ...  \n",
       "50871  0.327958    0.287360  0.004323  \n",
       "50872  0.327958    0.305231  0.001853  \n",
       "50873  0.330543    0.300580  0.001924  \n",
       "50874  0.330543    0.289464  0.002272  \n",
       "50875  0.330543    0.289473  0.002463  \n",
       "\n",
       "[42637 rows x 9 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add full kmer profiles or just RMSD\n",
    "full = False\n",
    "if full:\n",
    "    data_ext = pd.concat([data_p, diffs], axis=1)\n",
    "else:\n",
    "    data_ext = data_p.copy()\n",
    "    data_ext['RMSD'] = rmsd\n",
    "data_ext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly_ID</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Chromosome_ID</th>\n",
       "      <th>Plasmid_ID</th>\n",
       "      <th>Chr_len</th>\n",
       "      <th>Plasmid_len</th>\n",
       "      <th>Chr_GC</th>\n",
       "      <th>Plasmid_GC</th>\n",
       "      <th>RMSD</th>\n",
       "      <th>Len_ratio</th>\n",
       "      <th>GC_abs_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_023703895.1_ASM2370389v1_genomic.fna.gz</td>\n",
       "      <td>KLEBSIELLA PNEUMONIAE</td>\n",
       "      <td>CP077812.1</td>\n",
       "      <td>CP077813.1</td>\n",
       "      <td>5167508</td>\n",
       "      <td>54288</td>\n",
       "      <td>0.575169</td>\n",
       "      <td>0.527888</td>\n",
       "      <td>0.001915</td>\n",
       "      <td>95.186929</td>\n",
       "      <td>0.047281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100133.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>189346</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.638836</td>\n",
       "      <td>0.002416</td>\n",
       "      <td>34.640029</td>\n",
       "      <td>0.044421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100134.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>128913</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.614430</td>\n",
       "      <td>0.003242</td>\n",
       "      <td>50.878895</td>\n",
       "      <td>0.068827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100135.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>104740</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.636500</td>\n",
       "      <td>0.002477</td>\n",
       "      <td>62.621262</td>\n",
       "      <td>0.046757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_024106315.1_ASM2410631v2_genomic.fna.gz</td>\n",
       "      <td>ACINETOBACTER BAUMANNII</td>\n",
       "      <td>NZ_CP115645.1</td>\n",
       "      <td>NZ_CP115646.1</td>\n",
       "      <td>3917468</td>\n",
       "      <td>104651</td>\n",
       "      <td>0.392244</td>\n",
       "      <td>0.425471</td>\n",
       "      <td>0.001895</td>\n",
       "      <td>37.433641</td>\n",
       "      <td>0.033227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50871</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014366.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>3125</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.287360</td>\n",
       "      <td>0.004323</td>\n",
       "      <td>929.586560</td>\n",
       "      <td>0.040598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50872</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014367.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>27068</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.305231</td>\n",
       "      <td>0.001853</td>\n",
       "      <td>107.320748</td>\n",
       "      <td>0.022727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50873</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098817.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>44138</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.300580</td>\n",
       "      <td>0.001924</td>\n",
       "      <td>55.353981</td>\n",
       "      <td>0.029963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50874</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098818.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>39504</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289464</td>\n",
       "      <td>0.002272</td>\n",
       "      <td>61.847256</td>\n",
       "      <td>0.041079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50875</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098819.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>22579</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289473</td>\n",
       "      <td>0.002463</td>\n",
       "      <td>108.207361</td>\n",
       "      <td>0.041071</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42637 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Assembly_ID                 Taxonomy  \\\n",
       "0      GCA_023703895.1_ASM2370389v1_genomic.fna.gz    KLEBSIELLA PNEUMONIAE   \n",
       "1      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "2      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "3      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "6      GCF_024106315.1_ASM2410631v2_genomic.fna.gz  ACINETOBACTER BAUMANNII   \n",
       "...                                            ...                      ...   \n",
       "50871   GCF_003595385.1_ASM359538v1_genomic.fna.gz    STAPHYLOCOCCUS AUREUS   \n",
       "50872   GCF_003595385.1_ASM359538v1_genomic.fna.gz    STAPHYLOCOCCUS AUREUS   \n",
       "50873  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "50874  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "50875  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "\n",
       "                     Chromosome_ID     Plasmid_ID  Chr_len  Plasmid_len  \\\n",
       "0                       CP077812.1     CP077813.1  5167508        54288   \n",
       "1      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100133.1  6558951       189346   \n",
       "2      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100134.1  6558951       128913   \n",
       "3      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100135.1  6558951       104740   \n",
       "6                    NZ_CP115645.1  NZ_CP115646.1  3917468       104651   \n",
       "...                            ...            ...      ...          ...   \n",
       "50871                NZ_CP014365.1  NZ_CP014366.1  2904958         3125   \n",
       "50872                NZ_CP014365.1  NZ_CP014367.1  2904958        27068   \n",
       "50873                NZ_CP098816.1  NZ_CP098817.1  2443214        44138   \n",
       "50874                NZ_CP098816.1  NZ_CP098818.1  2443214        39504   \n",
       "50875                NZ_CP098816.1  NZ_CP098819.1  2443214        22579   \n",
       "\n",
       "         Chr_GC  Plasmid_GC      RMSD   Len_ratio  GC_abs_diff  \n",
       "0      0.575169    0.527888  0.001915   95.186929     0.047281  \n",
       "1      0.683257    0.638836  0.002416   34.640029     0.044421  \n",
       "2      0.683257    0.614430  0.003242   50.878895     0.068827  \n",
       "3      0.683257    0.636500  0.002477   62.621262     0.046757  \n",
       "6      0.392244    0.425471  0.001895   37.433641     0.033227  \n",
       "...         ...         ...       ...         ...          ...  \n",
       "50871  0.327958    0.287360  0.004323  929.586560     0.040598  \n",
       "50872  0.327958    0.305231  0.001853  107.320748     0.022727  \n",
       "50873  0.330543    0.300580  0.001924   55.353981     0.029963  \n",
       "50874  0.330543    0.289464  0.002272   61.847256     0.041079  \n",
       "50875  0.330543    0.289473  0.002463  108.207361     0.041071  \n",
       "\n",
       "[42637 rows x 11 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# compute other pairwise features\n",
    "data_ext['Len_ratio'] = data_ext['Chr_len'].div(data_ext['Plasmid_len'])\n",
    "data_ext['GC_abs_diff'] = data_ext['Chr_GC'].add(data_ext['Plasmid_GC'] * -1).abs()\n",
    "data_ext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iter 10000\n",
      "iter 15000\n",
      "iter 20000\n",
      "iter 25000\n",
      "iter 30000\n",
      "iter 35000\n",
      "iter 40000\n",
      "41684 rows populated out of 41684\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly_ID</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Chromosome_ID</th>\n",
       "      <th>Plasmid_ID</th>\n",
       "      <th>Chr_len</th>\n",
       "      <th>Plasmid_len</th>\n",
       "      <th>Chr_GC</th>\n",
       "      <th>Plasmid_GC</th>\n",
       "      <th>RMSD</th>\n",
       "      <th>Len_ratio</th>\n",
       "      <th>GC_abs_diff</th>\n",
       "      <th>SRCC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_023703895.1_ASM2370389v1_genomic.fna.gz</td>\n",
       "      <td>KLEBSIELLA PNEUMONIAE</td>\n",
       "      <td>CP077812.1</td>\n",
       "      <td>CP077813.1</td>\n",
       "      <td>5167508</td>\n",
       "      <td>54288</td>\n",
       "      <td>0.575169</td>\n",
       "      <td>0.527888</td>\n",
       "      <td>0.001915</td>\n",
       "      <td>95.186929</td>\n",
       "      <td>0.047281</td>\n",
       "      <td>0.9044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100133.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>189346</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.638836</td>\n",
       "      <td>0.002416</td>\n",
       "      <td>34.640029</td>\n",
       "      <td>0.044421</td>\n",
       "      <td>0.9816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100134.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>128913</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.614430</td>\n",
       "      <td>0.003242</td>\n",
       "      <td>50.878895</td>\n",
       "      <td>0.068827</td>\n",
       "      <td>0.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_024662395.1_ASM2466239v1_genomic.fna.gz</td>\n",
       "      <td>BURKHOLDERIA GLUMAE</td>\n",
       "      <td>NZ_CP100131.1,NZ_CP100132.1</td>\n",
       "      <td>NZ_CP100135.1</td>\n",
       "      <td>6558951</td>\n",
       "      <td>104740</td>\n",
       "      <td>0.683257</td>\n",
       "      <td>0.636500</td>\n",
       "      <td>0.002477</td>\n",
       "      <td>62.621262</td>\n",
       "      <td>0.046757</td>\n",
       "      <td>0.9788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_024106315.1_ASM2410631v2_genomic.fna.gz</td>\n",
       "      <td>ACINETOBACTER BAUMANNII</td>\n",
       "      <td>NZ_CP115645.1</td>\n",
       "      <td>NZ_CP115646.1</td>\n",
       "      <td>3917468</td>\n",
       "      <td>104651</td>\n",
       "      <td>0.392244</td>\n",
       "      <td>0.425471</td>\n",
       "      <td>0.001895</td>\n",
       "      <td>37.433641</td>\n",
       "      <td>0.033227</td>\n",
       "      <td>0.8831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50871</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014366.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>3125</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.287360</td>\n",
       "      <td>0.004323</td>\n",
       "      <td>929.586560</td>\n",
       "      <td>0.040598</td>\n",
       "      <td>0.9132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50872</th>\n",
       "      <td>GCF_003595385.1_ASM359538v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS AUREUS</td>\n",
       "      <td>NZ_CP014365.1</td>\n",
       "      <td>NZ_CP014367.1</td>\n",
       "      <td>2904958</td>\n",
       "      <td>27068</td>\n",
       "      <td>0.327958</td>\n",
       "      <td>0.305231</td>\n",
       "      <td>0.001853</td>\n",
       "      <td>107.320748</td>\n",
       "      <td>0.022727</td>\n",
       "      <td>0.9643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50873</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098817.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>44138</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.300580</td>\n",
       "      <td>0.001924</td>\n",
       "      <td>55.353981</td>\n",
       "      <td>0.029963</td>\n",
       "      <td>0.9836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50874</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098818.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>39504</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289464</td>\n",
       "      <td>0.002272</td>\n",
       "      <td>61.847256</td>\n",
       "      <td>0.041079</td>\n",
       "      <td>0.9706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50875</th>\n",
       "      <td>GCF_039640675.1_ASM3964067v1_genomic.fna.gz</td>\n",
       "      <td>STAPHYLOCOCCUS CAPITIS</td>\n",
       "      <td>NZ_CP098816.1</td>\n",
       "      <td>NZ_CP098819.1</td>\n",
       "      <td>2443214</td>\n",
       "      <td>22579</td>\n",
       "      <td>0.330543</td>\n",
       "      <td>0.289473</td>\n",
       "      <td>0.002463</td>\n",
       "      <td>108.207361</td>\n",
       "      <td>0.041071</td>\n",
       "      <td>0.9768</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41684 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Assembly_ID                 Taxonomy  \\\n",
       "0      GCA_023703895.1_ASM2370389v1_genomic.fna.gz    KLEBSIELLA PNEUMONIAE   \n",
       "1      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "2      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "3      GCF_024662395.1_ASM2466239v1_genomic.fna.gz      BURKHOLDERIA GLUMAE   \n",
       "6      GCF_024106315.1_ASM2410631v2_genomic.fna.gz  ACINETOBACTER BAUMANNII   \n",
       "...                                            ...                      ...   \n",
       "50871   GCF_003595385.1_ASM359538v1_genomic.fna.gz    STAPHYLOCOCCUS AUREUS   \n",
       "50872   GCF_003595385.1_ASM359538v1_genomic.fna.gz    STAPHYLOCOCCUS AUREUS   \n",
       "50873  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "50874  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "50875  GCF_039640675.1_ASM3964067v1_genomic.fna.gz   STAPHYLOCOCCUS CAPITIS   \n",
       "\n",
       "                     Chromosome_ID     Plasmid_ID  Chr_len  Plasmid_len  \\\n",
       "0                       CP077812.1     CP077813.1  5167508        54288   \n",
       "1      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100133.1  6558951       189346   \n",
       "2      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100134.1  6558951       128913   \n",
       "3      NZ_CP100131.1,NZ_CP100132.1  NZ_CP100135.1  6558951       104740   \n",
       "6                    NZ_CP115645.1  NZ_CP115646.1  3917468       104651   \n",
       "...                            ...            ...      ...          ...   \n",
       "50871                NZ_CP014365.1  NZ_CP014366.1  2904958         3125   \n",
       "50872                NZ_CP014365.1  NZ_CP014367.1  2904958        27068   \n",
       "50873                NZ_CP098816.1  NZ_CP098817.1  2443214        44138   \n",
       "50874                NZ_CP098816.1  NZ_CP098818.1  2443214        39504   \n",
       "50875                NZ_CP098816.1  NZ_CP098819.1  2443214        22579   \n",
       "\n",
       "         Chr_GC  Plasmid_GC      RMSD   Len_ratio  GC_abs_diff    SRCC  \n",
       "0      0.575169    0.527888  0.001915   95.186929     0.047281  0.9044  \n",
       "1      0.683257    0.638836  0.002416   34.640029     0.044421  0.9816  \n",
       "2      0.683257    0.614430  0.003242   50.878895     0.068827  0.9658  \n",
       "3      0.683257    0.636500  0.002477   62.621262     0.046757  0.9788  \n",
       "6      0.392244    0.425471  0.001895   37.433641     0.033227  0.8831  \n",
       "...         ...         ...       ...         ...          ...     ...  \n",
       "50871  0.327958    0.287360  0.004323  929.586560     0.040598  0.9132  \n",
       "50872  0.327958    0.305231  0.001853  107.320748     0.022727  0.9643  \n",
       "50873  0.330543    0.300580  0.001924   55.353981     0.029963  0.9836  \n",
       "50874  0.330543    0.289464  0.002272   61.847256     0.041079  0.9706  \n",
       "50875  0.330543    0.289473  0.002463  108.207361     0.041071  0.9768  \n",
       "\n",
       "[41684 rows x 12 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fetch SRCCs; takes a few minutes due to large file size\n",
    "data_ext['SRCC'] = 0.0\n",
    "\n",
    "# first trim assembly IDs\n",
    "assemblies_trimmed = data_ext['Assembly_ID'].str[:-7].values\n",
    "\n",
    "fn = '/fs/nexus-projects/plasmids/plasmid_host_range_2024_summer_interns/expanded_dataset/correlation_table_v4.txt'\n",
    "index = 0\n",
    "populated = 0\n",
    "with open(fn, 'r') as f:\n",
    "    # save list of plasmids\n",
    "    plasmid_list = f.readline().split()\n",
    "    while True:\n",
    "        if index < 6800: # found in practice the first 6800 lines aren't present in data_ext, so this makes things faster\n",
    "            next(f)\n",
    "            index += 1\n",
    "            continue\n",
    "        line = f.readline()\n",
    "        if line is None or line == '':\n",
    "            break\n",
    "        assembly = line.split('\\t', 1)[0]\n",
    "        assembly = assembly[:-15]\n",
    "        # check for containment; assume all assemblies found\n",
    "        if assembly in assemblies_trimmed:\n",
    "            rows_found = data_ext[data_ext['Assembly_ID'].str[:-7] == assembly]\n",
    "            for i, plasmid in enumerate(rows_found['Plasmid_ID']):\n",
    "                if plasmid not in plasmid_list:\n",
    "                    # drop row if plasmid not found\n",
    "                    data_ext.drop(rows_found.iloc[i].name, inplace=True)\n",
    "                else:\n",
    "                    # if both assembly and plasmid are found, split line further to extract plasmid\n",
    "                    plasmid_idx = plasmid_list.index(plasmid)\n",
    "                    # if idx=1, we need 3 splits: 1 for assembly ID, 2 for ID + idx 0, 3 for ID + idx 0-1\n",
    "                    # idx+1 indexing works in last index edge case, -2 indexing doesn't\n",
    "                    srcc = line.split('\\t', plasmid_idx+2)[plasmid_idx+1]\n",
    "                    # update dataframe\n",
    "                    data_ext.loc[rows_found.iloc[i].name, 'SRCC'] = float(srcc)\n",
    "                    populated += 1\n",
    "        index += 1\n",
    "        if index % 5000 == 0:\n",
    "            print('iter', index)\n",
    "print(populated, 'rows populated out of', len(data_ext))\n",
    "data_ext"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "tfpt",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.20"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
